pkg/provider/azure/azure.go

```diff
@@ -11,7 +13,55 @@ func GetClouProviderCredentials(fixedCredentials map[string]string) credentials.
 		FixedCredentials:  fixedCredentials}
 }
 
-var DefaultCredentials = GetClouProviderCredentials(nil)
+var (
+	DefaultCredentials               = GetClouProviderCredentials(nil)
+	locationsSupportingResourceGroup = []string{
```

Comment:
DId you check if is there any api call to know region supporting RG?

Reply:
Otherwise LGTM

Reply:
i did try to find, but couldn't find any API for getting the list of all locations where resource groups are supported, so went with putting a list of good locations in the code

we also need such an API to handle the missing images issue

Reply:
through graph?